### PR TITLE
Update detail page

### PR DESCRIPTION
**Pull request by fanimator12**
Certain elements (date, event details, sponsor) should not be displayed if they are **null**.

### CODE REVIEW

File: ui/src/routes/lecture.tsx [hunk 1]
```diff
@@ -67,13 +67,19 @@ function Lecture() {
...
+                  </Title>
+                )}
+                {lecture.event_details != null && (
+                  <a href={lecture.event_details}>Event details (Indico)</a>
+                )}
+                {lecture.sponsor != null && (
+                  <Title level={4} id="sponsor">
+                    Sponsored by <strong>{lecture.sponsor}</strong>
+                  </Title>
+                )}
```
[the remark this answers is not in the document]
Comment (fanimator12) [reply ui/src/routes/lecture.tsx]: I updated it. It should be fine now.

File: ui/src/routes/lecture.tsx [hunk 2]
```diff
@@ -67,13 +67,21 @@ function Lecture() {
...
+                {lecture.event_details && lecture.event_details.length > 0 && (
+                  <a href={lecture.event_details} id="event">
+                    Event details (Indico)
+                  </a>
+                )}
+                {lecture.sponsor && lecture.sponsor.length > 0 && (
+                  <Title level={4} id="sponsor">
+                    Sponsored by <strong>{lecture.sponsor}</strong>
+                  </Title>
+                )}
```
Comment (drjova) [inline ui/src/routes/lecture.tsx:84]:
```suggestion
                {lecture.date && lecture.date !== '' > 0 && (
                  <Title level={4} id="date">
                    {lecture.date}
                  </Title>
                )}
                {lecture.event_details && lecture.event_details !== '' > 0 && (
                  <a href={lecture.event_details} id="event">
                    Event details (Indico)
                  </a>
                )}
                {lecture.sponsor && lecture.sponsor !== '' > 0 && (
                  <Title level={4} id="sponsor">
                    Sponsored by <strong>{lecture.sponsor}</strong>
                  </Title>
                )}
```

:)

Comment (fanimator12) [reply ui/src/routes/lecture.tsx:84]: @drjova you mean without `> 0`?

Comment (fanimator12) [reply ui/src/routes/lecture.tsx:84]: > :)

because this way it won't work

Comment (fanimator12) [reply ui/src/routes/lecture.tsx:84]: `{lecture.date && lecture.date !== '' && (
                  <Title level={4} id="date">
                    {lecture.date}
                  </Title>
                )}
                {lecture.event_details && lecture.event_details !== '' && (
                  <a href={lecture.event_details} id="event">
                    Event details (Indico)
                  </a>
                )}
                {lecture.sponsor && lecture.sponsor !== '' && (
                  <Title level={4} id="sponsor">
                    Sponsored by <strong>{lecture.sponsor}</strong>
                  </Title>
                )}
`
let me know if that's what you meant

Comment (drjova) [reply ui/src/routes/lecture.tsx:84]: without the `>0` sorry

Comment (fanimator12) [reply ui/src/routes/lecture.tsx:84]: Okay, I make a new commit then.

**Pull request merged**